### PR TITLE
Change pyproject.toml on Kedro project template to be compatible with `uv`, `rye`, `pdm`

### PR DESCRIPTION
**Pull request by lrcouto**
## Description

Changes the `pyproject.toml` file on the default project template to work with different package managers.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by astrojuanlu** [comment]
(Also don't forget https://github.com/kedro-org/kedro/issues/4116#issuecomment-2324868393 !)

File: kedro/templates/project/{{ cookiecutter.repo_name }}/pyproject.toml [hunk 1]
```diff
@@ -3,9 +3,16 @@ requires = ["setuptools"]
...
+dynamic = ["version"]
+dependencies = [
+    "ipython>=8.10",
+    "jupyterlab>=3.0",
+    "notebook",
+    "kedro~={{ cookiecutter.kedro_version }}"
+]
 
 [project.scripts]
 {{ cookiecutter.repo_name }} = "{{ cookiecutter.python_package }}.__main__:main"
```
Comment (merelcht) [inline kedro/templates/project/{{ cookiecutter.repo_name }}/pyproject.toml:18]: Should we add the quotes here as well to be consistent with starters (https://github.com/kedro-org/kedro-starters/blob/main/spaceflights-pandas/%7B%7B%20cookiecutter.repo_name%20%7D%7D/pyproject.toml#L11)?

**Review by merelcht — APPROVED** [review]
I would add this to the release notes as well for those users who are indeed using `uv`, `rye` or `pdm.

**Review by ElenaKhaustova — COMMENTED** [review]
Shall we also update test starter?

https://github.com/kedro-org/kedro/blob/main/features/steps/test_starter/%7B%7B%20cookiecutter.repo_name%20%7D%7D/pyproject.toml

**Review by ElenaKhaustova — APPROVED** [review]
LGTM 👍

**Pull request merged**